scripts/sbatch.py

```diff
@@ -0,0 +1,288 @@
+        for line in lines:
+            if "SBATCH" in line and header in line:
+                value = line.strip().split("--{}=".format(header))[-1]
+                header_dict[header] = value
+                break
+    return header_dict
+
+
+def process_mem_per_cpu(mem_per_cpu):
+    """Convert a mem_per_cpu string to float"""
+    if isinstance(mem_per_cpu, str):
+        if mem_per_cpu[-1].upper() == "G":  # 0.5G to 0.5
+            mem_per_cpu = float(mem_per_cpu[:-1])  # 521MB to 0.5
+        elif mem_per_cpu[-2:] == "MB":
+            mem_per_cpu = float(mem_per_cpu[:-2]) / MB_PER_GB
+    else:
+        sys.exit("undefined mem per cpu, please add unit G/MB if not included")
+    return mem_per_cpu
+
+
+def process_header_values(slurm_header_dict, slurm_conf, hpc):
+    """
+       Process header dict values
+       For maui, mem_per_cpu is None as mem is not calculated against billing
+    """
+    if not slurm_header_dict[SlurmHeader.ntasks.value]:  # None, set to default
+        slurm_header_dict[SlurmHeader.ntasks.value] = DEFAULT_NTASKS
+    else:  # not None, convert str to int
+        slurm_header_dict[SlurmHeader.ntasks.value] = int(
+            slurm_header_dict[SlurmHeader.ntasks.value]
+        )
+
+    if not slurm_header_dict[SlurmHeader.cpus_per_task.value]:
+        slurm_header_dict[SlurmHeader.cpus_per_task.value] = DEFAULT_CPUS_PER_TASK
+    else:
+        slurm_header_dict[SlurmHeader.cpus_per_task.value] = int(
+            slurm_header_dict[SlurmHeader.cpus_per_task.value]
+        )
+
+    if hpc == HPC.mahuika.value:
+        if not slurm_header_dict[SlurmHeader.mem_per_cpu.value]:
+            slurm_header_dict[SlurmHeader.mem_per_cpu.value] = get_default_mem_per_cpu(
+                slurm_conf
+            )
+        else:
+            slurm_header_dict[SlurmHeader.mem_per_cpu.value] = process_mem_per_cpu(
+                slurm_header_dict[SlurmHeader.mem_per_cpu.value]
+            )
+
+    return slurm_header_dict
+
+
+def get_hpc_conf(partition):
+    """Get hpc name and path to slurm.conf according to partition name in the slurm file header"""
+    if partition in [p.value for p in MauiPartition]:
+        slurm_conf = MAUI_SLUM_CONF  # dummy, not used
+        hpc = HPC.maui.value
+    elif partition in [p.value for p in MahuikaPartition]:
+        slurm_conf = MAHUIKA_SLURM_CONF
+        hpc = HPC.mahuika.value
+    else:
+        sys.exit("{} partition does not exit")
+    return hpc, slurm_conf
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("slurm_to_sbatch", help="path to slurm file to be sbatched")
+    parser.add_argument(
+        "--core_hour_json",
+        default="core_hours.json",
+        help="path to core hour json file getting from dashboard.db, default is core_hours.json",
+    )
+    parser.add_argument(
+        "--user", default=getpass.getuser(), help="the user who's sbatching the job"
+    )
+    args = parser.parse_args()
```

Comment:
Needs a catch all to get all other sbatch arguements.

Reply:
try adding a positional argument with nargs='*'